backend/helper_zero/views/search.py

```diff
@@ -13,24 +15,52 @@ class SearchView(viewsets.ViewSet):
     """
 
     def list(self, request):
-        delivery = request.query_params.get('delivery', False)
-        item_type = request.query_params.get('item_type')
-        lat = request.query_params.get('lat')
-        lon = request.query_params.get('lon')
-        radius = request.query_params.get('radius', 10)
-        zipcode = request.query_params.get('zipcode')
         offset = int(request.query_params.get('offset', 0))
-
-        lat = float(lat) if lat else None
-        lon = float(lon) if lon else None
-        delivery_values = [True] if delivery else [True, False]
-
-        org_search_results = _get_org_search_results(
-            zipcode, lat, lon, radius, delivery_values, offset
-        )
-
-        response = _filter_org_list_by_donation_requests(org_search_results, item_type)
-        return Response(response)
+        item_type = request.query_params.get('item_type')
+        location_params, org_filter_params, donation_request_filter_params = _process_params(request.query_params)
```

Comment:
if you have a helper function to read from query_params it might be useful to just receive `offset` and `item_type` from the `_process_params` function (cleaner that way since you're already pulling fields out there)